### ISSUE TITLE
typeof check for null

### ISSUE THREAD

**Issue by sabithpocker** [getify/You-Dont-Know-JS#552]
In [Chapter 1](https://github.com/getify/You-Dont-Know-JS/blob/master/types%20%26%20grammar/ch1.md) of Types & Grammar it is specified that:

If you want to test for a null value using its type, you need a compound condition:

    var a = null;
    (!a && typeof a === "object"); // true

I understand that its explaining the problem with `typeof` null, but the workaround specified doesn't seem right to me. 
It would be nice if we mention the best way of checking for `null` in Javascript at least as a side note. I have been thinking that this is the best way to check for `null` until I saw a [discussion](http://wiki.ecmascript.org/doku.php?id=proposals:typeof) between Douglas Rockford and Brendan Eich where this is mentioned as an idiotic approach.

PS: I have no idea on what is the best way to check for null, but guess its `a === null`.

**Comment by getify** [comment]
To be fair, doug calls lots things he doesn't like "idiotic" and the people who do them as "idiots".

I don't think it's idiotic. It's not ideal, for sure. And probably less performant. But the spirit of the section is to talk about use of `typeof` specifically. There are other parts of the text that talk about the virtues of `===` (and the downfalls, frankly). Of course `a === null` is more specific, but then it hides the fact that `typeof null` has a gotcha. I didn't make such a side-note about `=== null` here so as to not confuse the issue, not in an attempt to endorse the `typeof` approach as appropriate.

TBH, I'm not sure why someone would want to identify something specifically as `null` (and not `undefined`). I think that's a bad idea in most cases. The book advocates for using coercion (via `==`) to make it so that you don't care whether it's `null` or `undefined`. I think that's the better approach, better even than `=== null`.

**Comment by sabithpocker** [comment]
Hey,
That sounds rational. I have been going through these language specifics while creating a curriculum for Developers, that's why I had to worry about the best practices. Yes, in real scenarios there won't be a specific requirement to check for `null`, I got blinded with interview Questions and stuff :).  The book helped me a lot, [Thanks a lot for the team :)] so just thought of sending a review as I faced an issue.
Thank you for the time and effort. I'll close the issue as you feel this is not required :)
